### ISSUE TITLE
MacOS 'overflow when subtracting duration from instant' just after node start

### ISSUE THREAD

**Issue by vmarkushin** [near/nearcore#7347]
**Describe the bug**
When running an example in [workspaces-rs](https://github.com/near/workspaces-rs), a panic occurs:
```bash
 $ cargo run --example nft
    Finished dev [unoptimized + debuginfo] target(s) in 0.57s
     Running `target/debug/examples/nft`
thread 'actix-rt|system:0|arbiter:1' panicked at 'overflow when subtracting duration from instant', library/std/src/time.rs:424:33
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Error: Failed to connect to RPC service http://localhost:24699 within 10 seconds: TransportError(SendError(PayloadSendError(reqwest::Error { kind: Request, url: Url { scheme: "http", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("localhost")), port: Some(24699), path: "/", query: None, fragment: None }, source: hyper::Error(Connect, ConnectError("tcp connect error", Os { code: 61, kind: ConnectionRefused, message: "Connection refused" })) })))
```

The overflow happens [here](https://github.com/near/nearcore/blob/7efb57a06906b297937fc0596139bd3da85aafc7/chain/network/src/peer_manager/peer_manager_actor.rs#L472).
I believe the source of the problem is that the `Instant::now` std function uses [`mach_absolute_time`](https://developer.apple.com/documentation/kernel/1462446-mach_absolute_time) internally, which according to the documentation, can start "at an arbitrary point". In particular, sometimes it can start from ~0, which, together with the subtraction of `Duration::hours(1)`, leads to the overflow.

**To Reproduce**
Steps to reproduce the behavior.
On (M1?) mac (perhaps multiple times to catch the bug):
```bash
cargo run --example nft
```

**Version (please complete the following information):**
- nearcore: master
- rust version: 1.62.0
- local

**Additional context**
MacOS: 12.5 (21G72)
Chip: Apple M1 Max

**Comment by matklad** [comment]
closed by #7341

**Comment by matklad** [comment]
Also, this is arguable something std should help with: https://github.com/rust-lang/rust/issues/100141
